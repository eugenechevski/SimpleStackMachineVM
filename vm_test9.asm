	# $Id: vm_test9.asm,v 1.2 2024/08/22 03:03:33 leavens Exp $
	.text 0
	SRI $sp, 1
	XORI $sp, 0, 0x567a
	LIT $sp, -1, 0x67a
	BORI $sp, -1, 0x5000
	BNE $sp, -1, 8 # fail if not equal
	ANDI $sp, 0, 0xFFF
	LIT $sp, -2, 0x67a
	BNE $sp, -2, 5 # fail if not equal
good:   NOTR
	PSTR $gp, 0
	STRA
	EXIT 0
bad:	NOTR
	PSTR $gp, 1
	STRA
	EXIT 1
	.data 1024
	STRING[1] Y = "Y\n"
	STRING [1] no = "no\n"
	CHAR nl = '\n'
	.stack 4096
	.end
