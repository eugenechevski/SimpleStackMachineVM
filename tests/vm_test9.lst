Address Instruction
     0: SRI $sp, 1
     1: XORI $sp, 0, 0x567a
     2: LIT $sp, -1, 1658
     3: BORI $sp, -1, 0x5000
     4: BNE $sp, -1, 8	# target is word address 12
     5: ANDI $sp, 0, 0xfff
     6: LIT $sp, -2, 1658
     7: BNE $sp, -2, 5	# target is word address 12
     8: NOTR 
     9: PSTR $gp, 0
    10: STRA 
    11: EXIT 0
    12: NOTR 
    13: PSTR $gp, 1
    14: STRA 
    15: EXIT 1
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     

