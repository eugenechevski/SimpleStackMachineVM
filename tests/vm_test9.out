      PC: 0
GPR[$gp]: 1024 	GPR[$sp]: 4096 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     

    4096: 0	

==>      0: SRI $sp, 1
      PC: 1
GPR[$gp]: 1024 	GPR[$sp]: 4095 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     

    4095: 0	        ...     

==>      1: XORI $sp, 0, 0x567a
      PC: 2
GPR[$gp]: 1024 	GPR[$sp]: 4095 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     

    4095: 22138	    4096: 0	

==>      2: LIT $sp, -1, 1658
      PC: 3
GPR[$gp]: 1024 	GPR[$sp]: 4095 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     
    4094: 1658	
    4095: 22138	    4096: 0	

==>      3: BORI $sp, -1, 0x5000
      PC: 4
GPR[$gp]: 1024 	GPR[$sp]: 4095 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     
    4094: 22138	
    4095: 22138	    4096: 0	

==>      4: BNE $sp, -1, 8	# target is word address 12
      PC: 5
GPR[$gp]: 1024 	GPR[$sp]: 4095 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     
    4094: 22138	
    4095: 22138	    4096: 0	

==>      5: ANDI $sp, 0, 0xfff
      PC: 6
GPR[$gp]: 1024 	GPR[$sp]: 4095 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     
    4094: 22138	
    4095: 1658	    4096: 0	

==>      6: LIT $sp, -2, 1658
      PC: 7
GPR[$gp]: 1024 	GPR[$sp]: 4095 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     
    4093: 1658	    4094: 22138	
    4095: 1658	    4096: 0	

==>      7: BNE $sp, -2, 5	# target is word address 12
      PC: 8
GPR[$gp]: 1024 	GPR[$sp]: 4095 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     
    4093: 1658	    4094: 22138	
    4095: 1658	    4096: 0	

==>      8: NOTR 
Y
      PC: 11
GPR[$gp]: 1024 	GPR[$sp]: 4095 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 2649	    1025: 683886	    1026: 10	    1027: 0	        ...     
    4093: 1658	    4094: 22138	
    4095: 2	    4096: 0	

==>     11: EXIT 0
